test/e2e/scheduling/events.go

```diff
@@ -23,12 +23,15 @@ import (
 	"k8s.io/api/core/v1"
 )
 
-func scheduleSuccessEvent(podName, nodeName string) func(*v1.Event) bool {
+func scheduleSuccessEvent(ns, podName, nodeName string) func(*v1.Event) bool {
 	return func(e *v1.Event) bool {
 		return e.Type == v1.EventTypeNormal &&
 			e.Reason == "Scheduled" &&
 			strings.HasPrefix(e.Name, podName) &&
-			strings.Contains(e.Message, fmt.Sprintf("Successfully assigned %v to %v", podName, nodeName))
+			// Check event message for 1.11+ (1.11 included)
+			(strings.Contains(e.Message, fmt.Sprintf("Successfully assigned %v/%v to %v", ns, podName, nodeName)) ||
+				// Check event message for 1.10- (1.10 included)
+				strings.Contains(e.Message, fmt.Sprintf("Successfully assigned %v to %v", podName, nodeName)))
```

Comment:
Relying on message strings is always error prone (as the issue shows) as any message string can change anytime without knowing that the same string is being matched somewhere else.

Reply:
yes, but this's enough for an e2e test.

Reply:
I will LGTM for this temporary fix. But it would be great if we can work out a change in scheduler to remove these string checks later.

Also @k82cn, could you please add a NOTE comment  on in scheduler.go to clarify that any change of this event message `Successfully assigned ...`  will affect scheduling e2e tests?

Reply:
Yeah I was checking for event strings in my e2e tests, and finally I removed them after having to fix the strings countless times after they changed.

Is it sufficient in this test to just check the pod.nodename field?

Reply:
It's also a challenge dealing with this string differences in upgrade/downgrade tests

Reply:
> Is it sufficient in this test to just check the pod.nodename field?

Should be enough for e2e test, as we build cases in one namespace.

I'd like to follow the style of 1.10 in this PR, and enhance in other pr for upper version.

Reply:
As a short term test fix, this is fine.  Longer term, like @resouer mentioned, it's probably better for the test to not check for strings altogether.

Reply:
Great! let's take cherrypick as cherrypick .